packages/discord.js/src/managers/GuildMemberRoleManager.js

```diff
@@ -174,6 +160,45 @@ class GuildMemberRoleManager extends DataManager {
     }
   }
 
+  /**
+   * Modifies the roles of the member.
+   * @param {RoleResolvable|RoleResolvable[]|Collection<Snowflake, Role>} rolesToAdd The roles to add
+   * @param {RoleResolvable|RoleResolvable[]|Collection<Snowflake, Role>} rolesToRemove The roles to remove
+   * @param {string} [reason] Reason for modifying the roles
+   * @returns {Promise<GuildMember>}
+   */
+  async modify(rolesToAdd, rolesToRemove, reason) {
+    const resolvedRolesToAdd = this.resolveRoles(rolesToAdd);
+    const resolvedRolesToRemove = this.resolveRoles(rolesToRemove);
+
+    const currentRoles = new Set(this.member.roles.cache.keys());
+    for (const role of resolvedRolesToAdd) {
+      currentRoles.add(role.id);
+    }
+    for (const role of resolvedRolesToRemove) {
+      currentRoles.delete(role.id);
+    }
+
+    return this.member.roles.set([...currentRoles], reason);
+  }
+
+  /**
+   * Resolves roles from the input.
+   * @param {RoleResolvable[] | Collection<Snowflake, Role>} rolesToResolve The roles to resolve
+   * @returns {Array} The resolved roles
+   */
+  resolveRoles(rolesToResolve) {
```

Comment:
This would either need to be marked as a private method or added to typings if you plan on making this a public method, refer to suggestion above if you want to make it private.